test/e2e/framework/conformance/conformance.go

```diff
@@ -0,0 +1,940 @@
+		listNamespace = func(_ context.Context) ([]runtime.Object, error) {
+			return namespaceInformer.Lister().List(labels.Everything())
+		}
+		go namespaceInformer.Informer().RunWithContext(ctx)
+		informersHaveSynced = append(informersHaveSynced, namespaceInformer.Informer().HasSynced)
+	}
+	if !cache.WaitForNamedCacheSyncWithContext(ctx, informersHaveSynced...) {
+		ginkgo.Fail("informers should have synced and didn't")
+	}
+
+	// matchObject generates a matcher which checks the result of a list operation
+	// against the expected object. Content verification is optional. Without it,
+	// only the namespace and name are checked.
+	matchObject := func(expectedObject *unstructured.Unstructured) gtypes.GomegaMatcher {
+		return &matchObjectList{expectedObject: expectedObject, verifyContent: tc.VerifyContent()}
+	}
+	gomega.Expect(listResource(ctx)).Should(matchObject(nil), "initial list of resources from informer cache")
+	gomega.Expect(resourceEvents.list()).To(gomega.HaveField("Events", gomega.BeEmpty()), "no events from resource informer yet")
+	if listNamespace != nil {
+		gomega.Expect(listNamespace(ctx)).Should(matchObject(nil), "initial list of namespace from informer cache")
+		gomega.Expect(resourceEvents.list()).To(gomega.HaveField("Events", gomega.BeEmpty()), "no events from namespace informer yet")
+	}
+
+	// matchEvents generates a matcher which checks the informer event list.
+	//
+	// The events are expected to involve only the given object.
+	// The ResourceVersion must not decrease.
+	// The sequence of valid events is given as a regular expression
+	// which is applied to the string returned by [EventList.Types].
+	matchEvents := func(obj *unstructured.Unstructured, regexp string) gtypes.GomegaMatcher {
+		// Verify namespace/name/uid of ids event.
+		ids := gomega.HaveField("Events", gomega.Or(gomega.BeEmpty(), gomega.HaveEach(gomega.HaveField("ID()", gomega.Equal(fmt.Sprintf("%s, %s", klog.KObj(obj), obj.GetUID()))))))
+
+		// Match the regexp against the result of Types().
+		order := gomega.HaveField("Types()", gomega.MatchRegexp(regexp))
+
+		// Include full object dump, HaveField itself doesn't.
+		return framework.GomegaObject(gomega.And(ids, order))
+	}
+
+	// ResourceVersion must not decrease. Delete events reset the sequence.
+	var resourceRV, namespaceRV big.Int
+	nextEvents := func(rv *big.Int, events eventList) {
+		ginkgo.GinkgoHelper()
+
+		checkNext := func(obj any) {
+			ginkgo.GinkgoHelper()
+			if obj == nil {
+				return
+			}
+			if tomb, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+				obj = tomb.Obj
+			}
+			metaData, err := meta.Accessor(obj)
+			framework.ExpectNoError(err, "access meta data")
+			gomega.Expect(metaData).To(apimachineryutils.HaveValidResourceVersion())
+
+			nextRV, ok := new(big.Int).SetString(metaData.GetResourceVersion(), 10)
+			if !ok {
+				framework.Failf("Parse ResourceVersion %s as int", metaData.GetResourceVersion())
+			}
+			if nextRV.Cmp(rv) < 0 {
```

Comment:
don't parse and compare with big.Int here, keep them as strings, use the apimachinery method made for this purpose, and assert >= 0 and no error

https://github.com/kubernetes/kubernetes/blob/8ac5701d3a146b367c83449d1eb6d872c003752e/staging/src/k8s.io/apimachinery/pkg/util/resourceversion/resourceversion.go#L32-L45